### ISSUE TITLE
Cannot change SDVX shop name/region name

### ISSUE THREAD

**Issue by ghost** [DragonMinded/bemaniutils#38]
In SDVX the shop/region name is shown as 3 orange stars, but when running locally its blank, Is there a way to change this?
edit : its called the area name in the test menu

**Comment by DragonMinded** [comment]
I'm not actually sure. I know you can change the shop name in network options, but area should be based on the prefecture of the server. There isn't a config for that, maybe there should be?

**Comment by DragonMinded** [comment]
It looks like this is set by a combination of location/country and location/region in the facility.get message, which I'm unaware of any network handling properly. I'll figure it out and see if I can't get it settable.

**Comment by DragonMinded** [comment]
https://github.com/DragonMinded/bemaniutils/commit/d05c3f907dfa18abd634c0197db63378a0cc161a and https://github.com/DragonMinded/bemaniutils/commit/c7affef159c268f0f4b5c9f7e500cf88147f87cb combined should allow you to change the area for all games connected to your server or in your arcade. Give it a test and see if it works for you.

**Comment by ghost** [comment]
sqlalchemy.exc.OperationalError

sqlalchemy.exc.OperationalError: (MySQLdb._exceptions.OperationalError) (1054, "Unknown column 'pref' in 'field list'")
[SQL: SELECT name, description, pin, pref, data FROM arcade WHERE id = %s]
[parameters: (1,)]
(Background on this error at: https://sqlalche.me/e/14/e3q8) 

I'm getting this error once I open the frontend

**Comment by DragonMinded** [comment]
Sounds like you didn't read the README. `./dbutils --config config/server.yaml upgrade` should be run to upgrade your local schema when there are DB changes.

**Comment by DragonMinded** [comment]
Have you been able to confirm the updated functionality?

**Comment by ghost** [comment]
i've tried but in game it says that my region is "不明" (unknown) and in the test menu my area is "未取得" (not obtained). in the front end my location is set to Hokkaido. i am testing locally right now, haven't had the time to upload the new version into my server so idk if that changes anything 

also my apologies for replying late i was very busy with real life stuff..

**Comment by DragonMinded** [comment]
What version of SDVX does this? Some of them require a different mapping and its possible you're looking at one that wasn't corrected.

**Comment by ghost** [comment]
SDVX Heavenly Haven, 2019020600

**Comment by ghost** [comment]
would there be a way to choose a custom region name with this implementation?

**Comment by DragonMinded** [comment]
No, its a hardcoded list based on a few things in the facility packet.

**Comment by RinChanNOWWW** [comment]
The area will be set from the fields `regionname` and `regionjname` in the response packet of `facility.get`:

```xml
<response>
    <facility status="0">
        <location>
            ...
            <regionname __type="str">CUSTOM_AREA_NAME</regionname>
            <regionjname __type="str">CUSTOM_AREA_NAME</regionjname>
            ...
        </location>
            ...
</response>
```

You can customize your area name here.

**Comment by RinChanNOWWW** [comment]
You can add codes like:

```python
location.add_child(Node.string('regionname', 'CUSTOM_AREA_NAME'))
location.add_child(Node.string('regionjname', 'CUSTOM_AREA_NAME'))
```

somewhere near

https://github.com/DragonMinded/bemaniutils/blob/90f3d9e1389e76c2d2078bf14a1eb0f351558b5f/bemani/backend/core/core.py#L169-L176

**Comment by wz18207** [comment]
> You can add codes like:
> 
> ```python
> location.add_child(Node.string('regionname', 'CUSTOM_AREA_NAME'))
> location.add_child(Node.string('regionjname', 'CUSTOM_AREA_NAME'))
> ```
> 
> somewhere near
> 
> https://github.com/DragonMinded/bemaniutils/blob/90f3d9e1389e76c2d2078bf14a1eb0f351558b5f/bemani/backend/core/core.py#L169-L176

it changed in config/server.yaml
```# Default region for this network (set to USA by default). See RegionConstants
    # for details on acceptible values. The range of accepted values is 1-56 matching
    # the 56 normal regions found in RegionConstants, and 1000 for "Europe" and
    # 2000 for "Other".
    region: 56
``
so just change region in config.

**Comment by RinChanNOWWW** [comment]
> > You can add codes like:
> > ```python
> > location.add_child(Node.string('regionname', 'CUSTOM_AREA_NAME'))
> > location.add_child(Node.string('regionjname', 'CUSTOM_AREA_NAME'))
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > somewhere near
> > https://github.com/DragonMinded/bemaniutils/blob/90f3d9e1389e76c2d2078bf14a1eb0f351558b5f/bemani/backend/core/core.py#L169-L176
> 
> it changed in config/server.yaml
> 
> ```
>     # for details on acceptible values. The range of accepted values is 1-56 matching
>     # the 56 normal regions found in RegionConstants, and 1000 for "Europe" and
>     # 2000 for "Other".
>     region: 56
> ``
> so just change region in config.
> ```

Sure, but the original codes will not set the area name (e.g., 北海道) which is at the front of the shop name (e.g., ROUND 1). Only if  add fields `regionname` (and `regionjname`?) will change the area name.

![image](https://user-images.githubusercontent.com/33975039/154657072-5575cad3-bd99-4576-8546-821a0c693548.png)

Without the fields, the "AREA" will be "未取得" (which means "not obtained").

**Comment by DragonMinded** [comment]
Custom area should be fixed as of 85a3f08c781870384fa3ab39436f486da463f38f for games that support it, thanks to @RinChanNOWWW
